File: crossecore-csharp/Java/Java/Java_/SingleVariableDeclaration.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	
	public interface SingleVariableDeclaration 
	: VariableDeclaration
	{
		Modifier modifier
		{
			get;
			set;
		}
		bool varargs
		{
			get;
		set;
		}
		TypeAccess type
		{
			get;
			set;
		}
		OrderedSet<Annotation> annotations
		{
			get;
		
		}
		AbstractMethodDeclaration methodDeclaration
		{
			get;
			set;
		}
		CatchClause catchClause
		{
			get;
			set;
		}
		EnhancedForStatement enhancedForStatement
		{
			get;
			set;
		}
	}
}

File: crossecore-typescript/java_/SingleVariableDeclaration.ts
import {OrderedSet} from "ecore/OrderedSet";
import {Set} from "ecore/Set";
import {Bag} from "ecore/Bag";
import {TypeAccess} from "./TypeAccess";
import {VariableDeclaration} from "./VariableDeclaration";
import {EnhancedForStatement} from "./EnhancedForStatement";
import {CatchClause} from "./CatchClause";
import {Modifier} from "./Modifier";
import {AbstractMethodDeclaration} from "./AbstractMethodDeclaration";
import {Annotation} from "./Annotation";

export interface SingleVariableDeclaration
extends VariableDeclaration

{
	varargs:boolean;
	
	modifier:Modifier;
	type:TypeAccess;
	annotations: OrderedSet<Annotation>;
	methodDeclaration:AbstractMethodDeclaration;
	catchClause:CatchClause;
	enhancedForStatement:EnhancedForStatement;
	

}

